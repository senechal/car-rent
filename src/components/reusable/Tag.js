import { useCallback } from 'react';
import PropTypes from 'prop-types';
const Tag = ({ value, ...props }) => (
  <span className="rac-tag" data-testid="tag" {...props} data-tagvalue={value} />
);

Tag.propTypes = {
  value: PropTypes.string.isRequired,
};

const Group = ({ onClose, ...props }) => {
  const handleBubbleClick = useCallback((event) => {
    const { tagvalue: value } = event.target.dataset;
    if (value) {
      onClose(value);
    }
  }, [onClose]);
  return <section {...props} onClick={handleBubbleClick} />
};

Group.propTypes = {
  onClose: PropTypes.func,
};
Group.defaultProps = {
  onClose: Function.prototype,
};

Tag.Group = Group;

export default Tag;
