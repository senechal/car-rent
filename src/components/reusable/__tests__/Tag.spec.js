import { render, fireEvent } from '@testing-library/react';
import Tag from '../Tag';

const { Group } = Tag;

describe('Testing Tag Components', () => {
  describe('Testing Tag', () => {
    it('should render correctly', () => {
      const { container } = render(<Tag value="value">content</Tag>);
      expect(container).toMatchSnapshot();
    });
  });
  describe('Testing Group', () => {
    it('should render correctlys', () => {
      const { container } = render(<Group><Tag value="value">content</Tag></Group>);
      expect(container).toMatchSnapshot();
    });
    it('should call onClose when clicked', () => {
      const callback = jest.fn();
      const { getByTestId } = render(<Group onClose={callback}><Tag value="value">content</Tag></Group>);
      const tag = getByTestId('tag');
      fireEvent.click(tag);
      expect(callback).toBeCalledWith('value');
    });
  });
});