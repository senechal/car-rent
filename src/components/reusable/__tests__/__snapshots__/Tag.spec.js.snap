// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tag Components Testing Group should render correctlys 1`] = `
<div>
  <section>
    <span
      class="rac-tag"
      data-tagvalue="value"
      data-testid="tag"
    >
      content
    </span>
  </section>
</div>
`;

exports[`Testing Tag Components Testing Tag should render correctly 1`] = `
<div>
  <span
    class="rac-tag"
    data-tagvalue="value"
    data-testid="tag"
  >
    content
  </span>
</div>
`;
